src/CMakeLists.txt

```diff
@@ -777,3 +787,47 @@ endif() # NOT WIN32
 if ( NOT WIN32)
   add_subdirectory(test test)
 endif() # NOT WIN32
+
+find_package(CUDAToolkit REQUIRED)
+
+# TODO: Remove excess stuff
+add_library(
+  python-binding-dependency
+  EXCLUDE_FROM_ALL
+  shared_memory_manager.h
+  shared_memory_manager.cc
+  data_compressor.h
+  common.h
+  common.cc
+  restricted_features.h
+  tracer.h
+  classification.cc
+)
+
+target_link_libraries(
+  python-binding-dependency
+  PUBLIC
+  http-endpoint-library
+  grpc-endpoint-library
+  CUDA::cudart
+  triton-core-serverapi           # from repo-core
+  triton-core-serverstub          # from repo-core
+  triton-common-json              # from repo-common
+)
+
+
+set_target_properties(python-binding-dependency PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+target_compile_features(python-binding-dependency PRIVATE cxx_std_${TRITON_MIN_CXX_STANDARD})
+
+target_compile_definitions(
+    python-binding-dependency
+    PRIVATE TRITON_ENABLE_GPU=1
+    PRIVATE TRITON_MIN_COMPUTE_CAPABILITY=${TRITON_MIN_COMPUTE_CAPABILITY}
+  )
+
+# Add the -fPIC flag
+# set_target_properties(server-src-lib PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
```

Comment:
cleanup

Reply:
not gonna comment on every instance, but would do a second look over for unused code and `TODO`s remaining